sklearn/externals/funcsigs.py

```diff
@@ -20,8 +20,8 @@
 __all__ = ['BoundArguments', 'Parameter', 'Signature', 'signature']
 
 
-_WrapperDescriptor = type(type.__call__)
```

Comment:
Scikit-learn use `type(type.__call__)` to get the type `wrapper_descriptor` and `_MethodWrapper = type(all.__call__)` to get `method_wrapper`, according the comment in here: https://github.com/Daetalus/scikit-learn/blob/ab4d8467ccd1586701c7bd698cfeeff460987b34/sklearn/externals/funcsigs.py#L159 It could avoid infinite recursive or segfault.

In Pyston the type of `type.__call__` is `instancemethod`. So it will cause segfault. I change it to `+_WrapperDescriptor = type(bytearray.__add__)`, because in Pyston, this type is `wrapper_descriptor`, same as CPython. `_MethodWrapper = type(u'str'.__add__)` used to get `method_wapper`. Hope it can help.